lib/application.js

```diff
@@ -42,9 +41,14 @@ module.exports = class Application extends Emitter {
     this.middleware = [];
     this.subdomainOffset = 2;
     this.env = process.env.NODE_ENV || 'development';
-    this.context = Object.create(context);
-    this.request = Object.create(request);
-    this.response = Object.create(response);
+
+    this.Context = class AppContext extends Context {};
+    this.Request = class AppRequest extends Request {};
+    this.Response = class AppResponse extends Response {};
+
+    this.context = this.Context.prototype;
+    this.request = this.Request.prototype;
+    this.response = this.Response.prototype;
   }
```

Comment:
>  I do think we should either ditch the prototypes or ditch the classes there, both is a little confusing.

:+1: using classes and prototypes in the same place is weird to me. should be one or the other